[Bug] Build fails when with serde >= 1.0.220

### Component

serde

### What version of Alloy are you on?

1.0.30

### Operating System

macOS (Apple Silicon)

### Describe the bug

Ran `cargo update on my project` alloy is failing to build with:

```
error[E0599]: no variant or associated item named `deserialize` found for enum `serde_core::__private::content::Content` in the current scope
   --> /Users/luca/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/alloy-consensus-1.0.30/src/transaction/envelope.rs:162:24
    |
162 | #[derive(Clone, Debug, TransactionEnvelope)]
    |                        ^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `serde_core::__private::content::Content<'_>`
    |
    = note: this error originates in the derive macro `TransactionEnvelope` (in Nightly builds, run with -Z macro-backtrace for more info)
```

If I force downgrade serde to 1.0.219 it works fine. 

It's likely due to they new `serde_core` crate they added [1.0.220](https://github.com/serde-rs/serde/releases/tag/v1.0.220)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -137,7 +137,7 @@ tracing-subscriber = "0.3"
 tracing-futures = "0.2"
 
 # serde
-serde = { version = "1.0", default-features = false, features = [
+serde = { version = "1.0.223", default-features = false, features = [
     "derive",
     "alloc",
 ] }

--- a/crates/tx-macros/src/serde.rs
+++ b/crates/tx-macros/src/serde.rs
@@ -232,7 +232,10 @@ impl<'a> SerdeGenerator<'a> {
                 where
                     D: #serde::Deserializer<'de>,
                 {
-                    let content = #serde::__private::de::Content::deserialize(deserializer)?;
+                    let content = #serde::de::DeserializeSeed::deserialize(
+                        #serde::__private::de::ContentVisitor::new(),
+                        deserializer
+                    )?;
                     let deserializer =
                         #serde::__private::de::ContentRefDeserializer::<D::Error>::new(&content);
 

